cmd/hclspecsuite/test_file.go

```diff
@@ -240,36 +240,83 @@ func (r *Runner) decodeDiagnosticsBlock(block *hcl.Block) ([]*TestFileExpectDiag
 	return ret, diags
 }
 
-func (r *Runner) decodeRangeFromBody(body hcl.Body) (hcl.Range, hcl.Body, hcl.Diagnostics) {
-	type RawPos struct {
-		Line   int `hcl:"line"`
-		Column int `hcl:"column"`
-		Byte   int `hcl:"byte"`
+func (r *Runner) decodePosFromBody(body hcl.Body) (hcl.Pos, hcl.Diagnostics) {
+	pos := hcl.Pos{}
+	var diags hcl.Diagnostics
+
+	posBody, moreDiags := body.Content(testFilePosSchema)
+	diags = append(diags, moreDiags...)
+
+	val, moreDiags := posBody.Attributes["line"].Expr.Value(nil)
+	diags = append(diags, moreDiags...)
+
+	err := gocty.FromCtyValue(val, &pos.Line)
+	if err != nil {
+		diags = diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid line number",
+			Detail:   fmt.Sprintf("The line number must be an integer: %s", err),
+			Subject:  posBody.Attributes["line"].Expr.Range().Ptr(),
+		})
 	}
-	type RawRange struct {
-		From   RawPos   `hcl:"from,block"`
-		To     RawPos   `hcl:"to,block"`
-		Remain hcl.Body `hcl:",remain"`
+
+	val, moreDiags = posBody.Attributes["column"].Expr.Value(nil)
+	diags = append(diags, moreDiags...)
+
+	err = gocty.FromCtyValue(val, &pos.Column)
+	if err != nil {
+		diags = diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid column number",
+			Detail:   fmt.Sprintf("The column number must be an integer: %s", err),
+			Subject:  posBody.Attributes["column"].Expr.Range().Ptr(),
+		})
+	}
+
+	val, moreDiags = posBody.Attributes["byte"].Expr.Value(nil)
+	diags = append(diags, moreDiags...)
+
+	err = gocty.FromCtyValue(val, &pos.Byte)
+	if err != nil {
+		diags = diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid byte position",
+			Detail:   fmt.Sprintf("The byte position must be an integer: %s", err),
+			Subject:  posBody.Attributes["byte"].Expr.Range().Ptr(),
+		})
 	}
 
-	var raw RawRange
-	diags := gohcl.DecodeBody(body, nil, &raw)
+	return pos, diags
+}
+
+func (r *Runner) decodeRangeFromBody(body hcl.Body) (hcl.Range, hcl.Body, hcl.Diagnostics) {
+	var diags hcl.Diagnostics
+
+	rangeBody, remain, moreDiags := body.PartialContent(testFileRangeSchema)
+
+	diags = append(diags, moreDiags...)
+	if moreDiags.HasErrors() {
+		return hcl.Range{}, nil, diags
+	}
```

Comment:
You don't need this. `rangeBody.Blocks` returns a slice, so even if it's empty it won't crash later.

Reply:
`rangeBody` is a pointer though (`*hcl.BodyContent`).

Reply:
That's true 👍 You could potentially do something like:

```
diags = append(diags, moreDiags...)
if rangeBody == nil {
  return hcl.Range{}, nil, diags
}
```

That would allow you to keep checking any parts of the configuration that might have been valid (like if the `from` block was parsed successfully, but the `to` block wasn't). I'll leave this up to you though!